apps/api/src/app/workflows-v2/workflow.controller.e2e.ts

```diff
@@ -193,6 +193,116 @@ describe('Workflow Controller E2E API Testing', () => {
       expect(idsDeduplicated.size).to.be.equal(10);
     });
   });
+
+  describe('Promote Workflow Permutations', () => {
+    it('should promote by creating a new workflow in production environment with the same properties', async () => {
+      // Create a workflow in the development environment
+      const devWorkflow = await createWorkflowAndValidate('-promote-workflow');
+
+      // Switch to production environment and get its ID
+      await session.switchToProdEnvironment();
+      const prodEnvironmentId = session.environment._id;
+      await session.switchToDevEnvironment();
+
+      // Promote the workflow to production
+      const res = await session.testAgent.put(`${v2Prefix}/workflows/${devWorkflow._id}/promote`).send({
+        targetEnvironmentId: prodEnvironmentId,
+      });
+
+      expect(res.status).to.equal(200);
+
+      const prodWorkflow = res.body.data;
+
+      // Verify that the promoted workflow has a new ID but the same workflowId
+      expect(prodWorkflow._id).to.not.equal(devWorkflow._id);
+      expect(prodWorkflow.workflowId).to.equal(devWorkflow.workflowId);
+
+      // Check that all non-environment-specific properties are identical
+      const propertiesToCompare = ['name', 'description', 'tags', 'preferences', 'status', 'type', 'origin'];
+      propertiesToCompare.forEach((prop) => {
+        expect(prodWorkflow[prop]).to.deep.equal(devWorkflow[prop], `Property ${prop} should match`);
+      });
+
+      // Verify that steps are correctly promoted
+      expect(prodWorkflow.steps).to.have.lengthOf(devWorkflow.steps.length);
+      prodWorkflow.steps.forEach((prodStep, index) => {
+        const devStep = devWorkflow.steps[index];
+        expect(prodStep.stepUuid).to.not.equal(devStep.stepUuid, 'Step UUID should be different');
+        expect(prodStep.controlValues).to.deep.equal(devStep.controlValues, 'Step controlValues should match');
+        expect(prodStep.name).to.equal(devStep.name, 'Step name should match');
+        expect(prodStep.type).to.equal(devStep.type, 'Step type should match');
+      });
+    });
+
```

Comment:
I guess we should not allow promotion in the same environment, right? if yes, let's add the test and validation